xmake/plugins/lua/xmake.lua

```diff
@@ -56,8 +61,33 @@ task("lua")
                 import("scripts." .. name).main(unpack(option.get("arguments") or {}))
             end
         else
+            local replhistory
+            if enable_readline then
+                -- clear history
+                readline.clear_history()
+
+                -- load history
+                replhistory = history.load("replhistory") or {}
+                for _, ln in ipairs(replhistory) do
+                    readline.add_history(ln)
+                end
+            end
+
             -- enter interactive mode
             sandbox.interactive()
+
```

Comment:
Why not handle them in `sandbox.interactive()`?  I don't want to expose too many interfaces that can't cross the platform in sandbox.